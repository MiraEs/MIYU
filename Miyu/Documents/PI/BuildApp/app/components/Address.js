'use strict';
import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	View,
	ViewPropTypes,
} from 'react-native';
import PhoneHelper from '../lib/PhoneHelper';
import { Text } from 'BuildLibrary';

class Address extends Component {
	renderAddressLine(address) {
		if (address) {
			return <Text>{address}</Text>;
		}
	}

	render() {
		const { address, boldName } = this.props;
		const weight = boldName ? 'bold' : 'normal';

		if (!address) {
			return null;
		}

		return (
			<View
				key={address.addressId}
				style={this.props.style}
			>
				<Text
					capitalize="all"
					weight={weight}
				>
					{address.firstName} {address.lastName}
				</Text>
				{this.renderAddressLine(address.company)}
				{this.renderAddressLine(address.address)}
				{this.renderAddressLine(address.address2)}
				<Text>{address.city}, {address.state} {address.zip}</Text>
				{this.renderAddressLine(PhoneHelper.formatPhoneNumber({ phoneNumber: address.phone }))}
			</View>
		);
	}
}

Address.propTypes = {
	address: PropTypes.shape({
		firstName: PropTypes.string.isRequired,
		lastName: PropTypes.string.isRequired,
		company: PropTypes.string,
		address: PropTypes.string,
		address2: PropTypes.string,
		city: PropTypes.string.isRequired,
		state: PropTypes.string.isRequired,
		zip: PropTypes.string.isRequired,
		phone: PropTypes.string,
	}),
	boldName: PropTypes.bool,
	style: ViewPropTypes.style,
};

Address.defaultProps = {
	boldName: true,
};

export default Address;
