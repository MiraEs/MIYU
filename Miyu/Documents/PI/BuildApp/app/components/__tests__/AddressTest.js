'use strict';

jest.mock('../../../app/services/httpClient', () => ({}));
jest.mock('../../../app/store/configStore', () => ({}));
jest.mock('../../../app/lib/analytics/tracking');
jest.mock('BuildNative');
jest.mock('BuildLibrary');

jest.unmock('react-native');

import Address from '../Address';
import React from 'react';

const defaultProps = {

};

describe('AddToCartButton component', () => {
	it('should render correctly', () => {
		const tree = require('react-test-renderer').create(
			<Address {...defaultProps} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

});
