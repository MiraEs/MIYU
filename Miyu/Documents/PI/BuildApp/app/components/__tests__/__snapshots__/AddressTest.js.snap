// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartButton component should render correctly 1`] = `null`;
